Include/Rover/Evaluator.hpp

```diff
@@ -1,9 +1,43 @@
 #ifndef ROVER_EVALUATION_HPP
 #define ROVER_EVALUATION_HPP
 #include <memory>
+#include <typeindex>
+#include <typeinfo>
+#include <type_traits>
+#include <unordered_map>
```

Comment:
Ok, but anyway: is there any reason to use a vector here?

Reply:
Performance reasons. ```unordered_map``` is notoriously inefficient.

Reply:
Another quirk of my mindset I guess. From my point, small size optimizations are bad for general-purpose applications because if the size of a structure is small, there is no need for any optimization in the first place. That's why I tend to interpret any introduction of theoretically sub-optimal complexity as a mistake.